src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs

```diff
@@ -172,43 +172,7 @@ private static string Replace(MatchEvaluator evaluator, Regex regex, string inpu
 
             (SegmentStringBuilder segments, MatchEvaluator evaluator, int prevat, string input, int count) state = (segments: SegmentStringBuilder.Create(), evaluator, prevat: 0, input, count);
 
-            if (regex._srm is not null)
-            {
-                Match match = regex.Match(input, startat, input.Length - startat);
-                if (!match.Success)
-                {
-                    return input;
-                }
-
-                while (match.Success)
-                {
-                    if (state.prevat < match.Index)
-                    {
-                        state.segments.Add(state.input.AsMemory(state.prevat, match.Index - state.prevat));
-                    }
-                    state.prevat = match.Index + match.Length;
-                    state.segments.Add(state.evaluator(match).AsMemory());
-
-                    if (state.count > 0)
-                    {
-                        state.count -= 1;
-                    }
-
-                    if (state.count == 0)
-                    {
-                        break;
-                    }
-
-                    match = match.NextMatch();
-                }
-
-                // final segment
-                if (state.prevat < input.Length)
-                {
-                    state.segments.Add(state.input.AsMemory(state.prevat, input.Length - state.prevat));
-                }
-            }
-            else if (!regex.RightToLeft)
+            if (!regex.RightToLeft)
```

Comment:
Aside, I notice SRM doesn't support RTL | EcmaScript. Do we expect to ship that way? Although my care level is low as it's likely very rarely used.

I also noticed we today don't seem to have any tests with RTL | EcmaScript, although I would expect the combination is unlikely to throw up any bugs.

Reply:
> although I would expect the combination is unlikely to throw up any bugs

To be clear,  it doesn't support either option,  not just both together. 

I think it's fine to ship that way.  We can always add in support later if it's really deemed impactful.

Reply:
On a related note, conceptually EcmaScript would be really easy to add. What it essentially affects is the meaning of \w and \d (probably also \s), all in a very systematic manner. In all those cases it is simply the intersection of the involved BDDs with the BDD for ascii and this happens when the BDDs are created from the original regex parse tree. 
(May be there are corner cases I don't understand. Not sure exactly what happens with \W and \D etc).

Reply:
As I recall from when we did some data mining in the past, even EcmaScript without RTL is pretty rare. I've not seen it in code either, eg., when folks open bugs. So I'd say this is fine to defer in favor of more impactful work.